File: Vibe.Domain/SupportRequests/SupportMessages/SupportMessage.cs
namespace Vibe.Domain.SupportRequests.SupportMessages
{
    public class SupportMessage
    {
        public Guid Id { get; }
        public String Text { get; }
        public DateTime CreatedAt { get; }
        public DateTime? ModifiedAt { get; }
        public Guid CreatedBy { get; }

        public SupportMessage(Guid id, string text, DateTime createdAt, DateTime? modifiedAt, Guid createdBy)
        {
            Id = id;
            Text = text;
            CreatedAt = createdAt;
            ModifiedAt = modifiedAt;
            CreatedBy = createdBy;
        }
    }
}

File: Vibe.Backoffice/vibe.backoffice.client/src/domain/techSupports/supportMessage.ts
export class SupportMessage {
    constructor(
        public id: string,
        public text: string,
        public createdAt: Date,
        public modifiedAt: Date | null,
        public createdBy: string
    ) {}
}

export function mapToSupportMessage(data: any): SupportMessage {
    const createdAt = new Date(data.createdAt)
    const modifiedAt = data.modifiedAt != null ? new Date(data.modifiedAt) : null

    return new SupportMessage(data.id, data.text, createdAt, modifiedAt, data.createdBy)
}

export function mapToSupportMessages(data: any[]): SupportMessage[] {
    return data.map(d => mapToSupportMessage(d))
}